traffic_ops/traffic_ops_golang/servercheck/servercheck.go

```diff
@@ -239,6 +237,35 @@ func DeprecatedReadServersChecks(w http.ResponseWriter, r *http.Request) {
 
 func handleReadServerCheck(inf *api.APIInfo, tx *sql.Tx) ([]tc.GenericServerCheck, error, error, int) {
 	extensions := make(map[string]string)
+
+	// Query Parameters to Database Query column mappings
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"id":   dbhelpers.WhereColumnInfo{"servercheck.server", api.IsInt},
+		"name": dbhelpers.WhereColumnInfo{"server.host_name", nil},
+	}
+
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(inf.Params, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		return nil, util.JoinErrs(errs), nil, http.StatusBadRequest
+	}
+
+	// where clause is different for servercheck and server table. Also, it differs for the query param.
+	var whereSC, whereSI string
+	if where != "" {
+		for k, _ := range inf.Params {
+			if k == "id" {
+				whereSC = where
+				whereSI = "WHERE (type.name LIKE 'MID%' OR type.name LIKE 'EDGE%') AND server.id=:id "
+			} else {
+				whereSC = ""
+				whereSI = "WHERE (type.name LIKE 'MID%' OR type.name LIKE 'EDGE%') AND server.host_name=:name "
+			}
```

Comment:
It looks like if I give both the actual filtering used here will be random. I can't imagine why I'd want to, but if I did a request for `serverchecks?id=1&hostName=foo` where server \#1 is *not* 'foo', I'd expect no results returned. Based on this code, I think it would probably return results for server \#1, and if I requested `serverchecks?hostName=foo&id=1` I think it'd probably return a _different_ result set - specifically the checks for the 'foo' server - even though I passed the exact same parameters.

I think it'd probably be best to check for each supported parameter's existence and add to the `where` clause appropriately based on whether it already has contents in turn